parts/k8s/kuberneteswindowssetup.ps1

```diff
@@ -40,7 +40,11 @@ param(
 
     [parameter(Mandatory=$true)]
     [ValidateNotNullOrEmpty()]
-    $AADClientSecret # base64
+    $AADClientSecret, # base64
+
+    [parameter(Mandatory=$true)]
+    [ValidateNotNullOrEmpty()]
+    $TargetEnvironment

```

Comment:
Should this be required for Azure deployments as well?

Reply:
The linux deploy process always puts the target environment in the config file. If it is left blank it defaults to AzurePublicCloud, so providing it doesn't make a difference for azure deployments.

Reply:
It is OK with me if this change does not break or alter behavior on Azure.
What does the `ValidateNotNullOrEmpty` attribute do if left black as you mention?

Reply:
The arguments for this script are contained in the ARM template in pkg/engine/vmextensions.go. TargetEnvironment is already a parameter in the template so on line 408 I pass it as an argument to this script. That means it will always be set to whatever the target environment is, be it AzureStackCloud or AzurePublicCloud

Reply:
Also, from what I can see leaving it blank could cause failures in China or Germany or the government cloud. It should be set to AzureChinaCloud or whichever so that the kubelet can connect correctly in those locations. It gets set every time when we deploy on a linux host but never gets set on windows hosts